plugins/utils/ccc.nix

```diff
@@ -5,50 +5,218 @@
   pkgs,
   ...
 }:
-helpers.neovim-plugin.mkNeovimPlugin config {
-  name = "ccc";
-  originalName = "ccc.nvim";
-  defaultPackage = pkgs.vimPlugins.ccc-nvim;
-
-  maintainers = [helpers.maintainers.JanKremer];
-
-  settingsOptions = {
-    default_color = helpers.defaultNullOpts.mkStr "#000000" ''
-      The default color used when a color cannot be picked. It must be HEX format.
-    '';
-
-    highlight_mode = helpers.defaultNullOpts.mkEnum ["fg" "bg" "foreground" "background"] "bg" ''
-      Option to highlight text foreground or background. It is used to
-      `output_line` and `highlighter`.
-    '';
-
-    lsp = helpers.defaultNullOpts.mkBool true ''
-      Whether to enable LSP support. The color information is updated in the
-      background and the result is used by `:CccPick` and highlighter.
-    '';
-
-    highlighter = {
-      auto_enable = helpers.defaultNullOpts.mkBool false ''
-        Whether to enable automatically on `BufEnter`.
+with lib;
+  helpers.neovim-plugin.mkNeovimPlugin config {
+    name = "ccc";
+    originalName = "ccc.nvim";
+    defaultPackage = pkgs.vimPlugins.ccc-nvim;
+
+    maintainers = [helpers.maintainers.JanKremer];
+
+    settingsOptions = let
+      listOfRawLua = with helpers.nixvimTypes; nullOr (maybeRaw (listOf strLua));
+      mapToRawLua = v:
+        if isList v
+        then map helpers.mkRaw v
+        else v;
+    in {
+      default_color = helpers.defaultNullOpts.mkStr "#000000" ''
+        The default color used when a color cannot be picked. It must be HEX format.
       '';
 
+      inputs = mkOption {
+        type = listOfRawLua;
+        default = [
+          "ccc.input.rgb"
+          "ccc.input.hsl"
+          "ccc.input.cmyk"
+        ];
```

Comment:
Why are we setting an explicit default? If this is important we should add a comment why

Reply:
[Looking upstream](https://github.com/uga-rosa/ccc.nvim/blob/f388f1981d222967c741fe9927edf9ba5fa3bcbe/doc/ccc.txt#L180-L203), that's the default used there.

My understanding is that we should be using a [default-null option](https://github.com/nix-community/nixvim/blob/c05fba47c733445098847399b61ddfd81ecbde55/lib/options.nix#L60-L72), which document upstream's default, as used when _our_ option is null. I.e. the `default` passed to `lib.mkOption` should be `null`.

Reply:
Yes exactly